build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/webpackTask.ts

```diff
@@ -56,22 +51,60 @@ export class WebpackTask extends LeafWithDoneFileTask {
 			}
 
 			return JSON.stringify(content);
-		} catch {
+		} catch (e) {
+			this.traceExec(`error generating done file content ${e}`);
 			return undefined;
 		}
 	}
 
 	private get configFileFullPath() {
 		// TODO: parse the command line for real, split space for now.
 		const args = this.command.split(" ");
-		let configFile = "webpack.config.js";
-		for (let i = 0; i < args.length; i++) {
+		for (let i = 1; i < args.length; i++) {
 			if (args[i] === "--config" && i + 1 < args.length) {
-				configFile = args[i + 1];
-				break;
+				return path.join(this.package.directory, args[i + 1]);
 			}
 		}
 
-		return path.join(this.package.directory, configFile);
+		return this.getDefaultConfigFile();
+	}
+
+	private getDefaultConfigFile() {
+		const defaultConfigFileNames = [
+			"webpack.config",
+			".webpack/webpack.config",
+			".webpack/webpackfile",
+		];
+		// TODO: webpack support more default config file extensions.  Just implement the ones that we use.
+		const defaultConfigExtensions = [".js", ".cjs"];
+		for (const name of defaultConfigFileNames) {
+			for (const ext of defaultConfigExtensions) {
+				const file = path.join(this.package.directory, `${name}${ext}`);
+				if (fs.existsSync(file)) {
+					return file;
+				}
+			}
+		}
+		// return webpack.config.js if nothing exist
+		return path.join(this.package.directory, "webpack.config.js");
+	}
+
+	private getEnvArguments() {
+		// TODO: parse the command line for real, split space for now.
+		const args = this.command.split(" ");
+		const env = {};
+		// Ignore trailing --env
+		for (let i = 1; i < args.length - 1; i++) {
+			if (args[i] == "--env") {
+				const value = args[++i].split("=");
+				env[value[0]] = value.length === 1 ? true : value[1];
+			}
+		}
+	}
+
+	private async getVersion() {
+		// TODO:  We can get webpack version with "webpack --version", but harder to get the plug-ins
+		// For now we just use the big hammer of the monorepo lock file as are guard against version change
+		return this.node.getLockFileHash();
```

Comment:
So this means that for now any change in the lockfile, whether webpack-related or not, will cause re-running of the webpack tasks?

Reply:
yes. unfortunately.   Alternative is that we can play it loose and just assume nothing change, and have to user notice it and do a rebuild, but those could be hard to diagnose.   I think it is better play it safe unless we found a way to get detail dependency information.

Reply:
@alexvy86 this could be improved by recording the installed version of dependencies of the packages (and transitive ones too) to scope it down a little bit.  Not sure if it is worth it.  It depends on how often the lock file change that doesn't cause a rebuild.